ocaml/xapi/xapi_pif.ml

```diff
@@ -221,12 +221,14 @@ let assert_no_protection_enabled ~__context ~self =
                   (Api_errors.redo_log_is_enabled, []))
   end
 
-let assert_not_sriov_logical_pif ~__context ~self =
+let assert_no_sriov ~__context ~self =
   let pif_rec = Db.PIF.get_record ~__context ~self in
-  match Xapi_pif_helpers.get_pif_type pif_rec with
-  | Network_sriov_logical _ ->
-    raise Api_errors.(Server_error (internal_error,
-        [Printf.sprintf "Can not forget network SR-IOV logical PIF: %s" (Ref.string_of self)]))
+  let topo = Xapi_pif_helpers.get_pif_topo ~__context ~pif_rec in
+  match topo, pif_rec.API.pIF_sriov_physical_PIF_of with
+  | Network_sriov_logical _ :: _, _
+  | VLAN_untagged _ :: Network_sriov_logical _ :: _, _
```

Comment:
You don't need this, because this case is covered by `assert_no_vlans`.

Reply:
Fixed.